### ISSUE TITLE
Inconsistency between `Grasp_Head_Local_Interaction`, label matching loss, and `pred_decode` for angle/depth extra class

### ISSUE THREAD

**Issue by rcao-hk** [iSEE-Laboratory/EconomicGrasp#16]
Hi, thanks for releasing the code.

I noticed a possible inconsistency between the angle/depth head definition, the label matching loss, and the decoding logic.

### Observation

In `Grasp_Head_Local_Interaction`, the angle and depth heads are defined with an extra output channel:

```python
self.conv_angle = nn.Conv1d(64, num_angle + 1, 1)
self.conv_depth = nn.Conv1d(64, num_depth + 1, 1)
```

So the prediction shapes are effectively:

* `grasp_angle_pred`: `(B, num_angle + 1, M)`
* `grasp_depth_pred`: `(B, num_depth + 1, M)`

However, in `pred_decode`, all channels are directly used in `argmax`:

```python
grasp_angle_pred = end_points['grasp_angle_pred'][i].float()
grasp_angle, grasp_angle_indxs = torch.max(grasp_angle_pred.squeeze(0), 0)
grasp_angle = grasp_angle_indxs * np.pi / 12

grasp_depth_pred = end_points['grasp_depth_pred'][i].float()
grasp_depth, grasp_depth_indxs = torch.max(grasp_depth_pred.squeeze(0), 0)
grasp_depth = (grasp_depth_indxs + 1) * 0.01
```

This means the last extra channel is also considered during decoding. 

### Why this seems inconsistent

From `process_grasp_labels`, the default labels are initialized as:

* `top_grasp_rotations = 12`
* `top_grasp_depth = 4`

but the final angle/depth losses are only computed on `valid_mask`:

```python
loss = criterion(grasp_angle_pred, grasp_angle_label)
loss = loss[valid_mask].mean()
```

and similarly for depth. 

So for invalid/unmatched seeds:

* the extra class index exists in the label tensor,
* but those samples are masked out by `valid_mask`,
* therefore the extra class is never used as a positive supervision target.

For valid samples, CE still suppresses that extra class as a non-target class, but it is never explicitly trained as a meaningful “invalid” class. 

### Question

Should the last extra channel be excluded during decoding, for example:

```python
angle_logits_valid = grasp_angle_pred[:num_angle, :]
depth_logits_valid = grasp_depth_pred[:num_depth, :]
```

before `argmax`?

Or is the intended design that the extra class should remain in decoding as well?

### Why I am asking

Currently, the implementation seems to mix two interpretations:

1. `num_angle + 1` / `num_depth + 1` suggests an extra reserved class exists.
2. The loss does not explicitly train this extra class as a valid semantic class for invalid samples.
3. `pred_decode` still allows this extra class to win the `argmax`.

This makes the role of the last channel unclear.

Thanks.

**Comment by DravenALG** [comment]
Yes, you are right. And excluding the last channel will be more stable. I will fix the code then.

Thanks for this!
